### ISSUE TITLE
[bug] XO TS config parserOptions.project does not allow xo to be ran outside of dir

### ISSUE THREAD

**Issue by Spence-S** [xojs/xo#580]
related to: https://github.com/Spence-S/vscode-linter-xo/issues/5 

Basically, xo uses the parserOption.project for use with typescript and it auto generates the tsconfig. However, it generates the tsconfig relative to where xo is called, and doesn't allow it to search from cwd argument.

This is all based off of a `cacheLocation` variable found here: https://github.com/Spence-S/xo/blob/f038f9c3b78430be829250b9438533aa5355bd74/lib/options-manager.js#L52

If we make the `cacheLocation` variable a function that takes a cwd argument, we can remedy this with no changes in behaviour while allowing both the cli and the api to be ran correctly on TS files outside of the current directory, which is a must for an xo editor extension for any editor.

**TL;DR**: xo needs to cache tsconfig relative to cwd option to work successfully in editor extensions.

**Comment by sindresorhus** [comment]
Good find. That was probably just an oversight. Pull request welcome :)

**Comment by daiyam** [comment]
@Spence-S Great! That might be the same issue that I'm just getting with `lint-staged`.
I'm using a lib that needed a custom type (`declare module...`), first commit, no issue.
Second commit, `lint-staged` is generating the errors (`no-unsafe...`) on the modification of the first commit. It looks like `xo`(via `lint-staged`) isn't finding the custom type... `npx xo` isn't generating a single issue.

**Comment by daiyam** [comment]
So far, `v0.43.0` hasn't fixed the issue with `lint-staged`.
For the extension `vscode-linter-xo`, I will wait for the update maybe after it has moved to XO... 👍 

Anyway, I will open an new issue for `lint-stage` and I will try to look around to find what causing the issue...

**Comment by Spence-S** [comment]
@daiyam you likely need to pass a cwd flag to fix the issue on lintstaged like 
`'*.ts': 'xo --fix --cwd="path/to/folder"'`

**Comment by daiyam** [comment]
I'm still getting the errors with `'*.ts': 'xo --cwd=src'` and `xo.config.js` moved into `src`.

All that because I want to use `import pkg from '../package.json';` and not do a `const pkg = require('../package.json');`... maybe I'm screwing the typescript config.

**Comment by Spence-S** [comment]
@daiyam ok I see - I checkout out your rept - the issue is closely tied to this and has to do with the way xo handles caching the tsconfig still. I am going to open another issue here. This fix was a step in the right direction though.
